### ISSUE TITLE
Web: Cross module dependency resolution failed

### ISSUE THREAD

**Issue by rafsanjani** [JetBrains/compose-multiplatform#3486]
**Describe the bug**
java.lang.IllegalArgumentException: Cross module dependency resolution failed due to signature 'org.w3c.dom.events/EventListener|2833721539368120484[0]' redefinition
	at org.jetbrains.kotlin.ir.backend.js.transformers.irToJs.JsIrModuleCrossModuleReferenceBuilder.buildCrossModuleRefs(JsIrProgramFragment.kt:168)
	at org.jetbrains.kotlin.ir.backend.js.transformers.irToJs.CrossModuleDependenciesResolver.resolveCrossModuleDependencies(JsIrProgramFragment.kt:131)
	at org.jetbrains.kotlin.ir.backend.js.transformers.irToJs.JsIrProgram.asCrossModuleDependencies(JsIrProgramFragment.kt:77)
	at org.jetbrains.kotlin.ir.backend.js.transformers.irToJs.IrModuleToJsTransformerKt.generateWrappedModuleBody(IrModuleToJsTransformer.kt:392)
	at org.jetbrains.kotlin.ir.backend.js.transformers.irToJs.IrModuleToJsTransformerKt.access$generateWrappedModuleBody(IrModuleToJsTransformer.kt:1)
	at org.jetbrains.kotlin.ir.backend.js.transformers.irToJs.JsCodeGenerator.generateJsCode(IrModuleToJsTransformer.kt:89)
	at org.jetbrains.kotlin.cli.js.K2JsIrCompiler$Ir2JsTransformer.compileAndTransformIrNew(K2JsIrCompiler.kt:140)

This works perfectly on **1.5.0-dev1136** and started breaking from **beta01**

**Affected platforms**
Select one of the platforms below:
- Web (K/JS) - Canvas based API

**Versions**
- Kotlin version*: 1.9.0 
- Compose Multiplatform version*: 1.5.0-beta02

**Comment by eymar** [comment]
It's a duplicate of https://github.com/JetBrains/compose-multiplatform/issues/3424

There is weird workaround:
> Moving implementation(compose.runtime) from under commonMain to under jsMain fixes the issue


It was fixed in kotlin - https://youtrack.jetbrains.com/issue/KT-60852/Kotlin-JSIR-IllegalArgumentException-Cross-module-dependency-resolution-failed-when-updating-to-Kotlin-1.9.0

**Comment by alexstyl** [comment]
> It's a duplicate of #3424
> 
> There is weird workaround:
> 
> > Moving implementation(compose.runtime) from under commonMain to under jsMain fixes the issue
> 
> It was fixed in kotlin - https://youtrack.jetbrains.com/issue/KT-60852/Kotlin-JSIR-IllegalArgumentException-Cross-module-dependency-resolution-failed-when-updating-to-Kotlin-1.9.0

I tried moving `implementation(compose.runtime)` to `jsMain` but I am still getting the same error

**Comment by ZacSweers** [comment]
> Moving implementation(compose.runtime) from under commonMain to under jsMain fixes the issue

This workaround doesn't appear to work for my case, continues with the same error

**Comment by eymar** [comment]
@ZacSweers Could you please try it with kotlin `1.9.10-456`, it's in dev repo - `maven("https://maven.pkg.jetbrains.space/kotlin/p/kotlin/dev/")`

You will need to supress the version check:
https://github.com/JetBrains/compose-multiplatform/blob/master/VERSIONING.md#disabling-kotlin-compatibility-check

```kotlin
compose {
    kotlinCompilerPlugin.set("1.5.0")
    kotlinCompilerPluginArgs.add("suppressKotlinVersionCompatibilityCheck=1.9.10-456")
}
```
it has to be added in modules with compose.
This would help to do that: https://github.com/JetBrains/compose-multiplatform/blob/test_cases_with_k2/compose/integrations/composable-test-cases/build.gradle.kts#L22

**Comment by eymar** [comment]
@ZacSweers I tried your branch with this patch and it works:

```
Index: gradle/libs.versions.toml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/gradle/libs.versions.toml b/gradle/libs.versions.toml
--- a/gradle/libs.versions.toml	(revision 06431d721baafe6d5839e36acc396721bd55bb73)
+++ b/gradle/libs.versions.toml	(date 1692281642182)
@@ -30,7 +30,7 @@
 dokka = "1.8.20"
 eithernet = "1.5.0"
 jdk = "20"
-kotlin = "1.9.0"
+kotlin = "1.9.10-456"
 kotlinpoet = "1.14.2"
 kotlinx-coroutines = "1.7.3"
 ksp = "1.9.0-1.0.13"
Index: samples/counter/apps/build.gradle.kts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/samples/counter/apps/build.gradle.kts b/samples/counter/apps/build.gradle.kts
--- a/samples/counter/apps/build.gradle.kts	(revision 06431d721baafe6d5839e36acc396721bd55bb73)
+++ b/samples/counter/apps/build.gradle.kts	(date 1692282209466)
@@ -30,7 +30,11 @@
   ios()
   js {
     moduleName = "counterapp"
-    browser()
+    browser {
+      commonWebpackConfig {
+        outputFileName = "counterapp.js"
+      }
+    }
     binaries.executable()
   }
   // endregion
Index: build.gradle.kts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/build.gradle.kts b/build.gradle.kts
--- a/build.gradle.kts	(revision 06431d721baafe6d5839e36acc396721bd55bb73)
+++ b/build.gradle.kts	(date 1692282097958)
@@ -144,6 +144,13 @@
     configureComposeBom(dependencies)
   }
 
+  afterEvaluate {
+    project.extensions.findByType<org.jetbrains.compose.ComposeExtension>()?.also {
+      it.kotlinCompilerPlugin.set("1.5.0")
+      it.kotlinCompilerPluginArgs.add("suppressKotlinVersionCompatibilityCheck=1.9.10-456")
+    }
+  }
+
   val hasCompose = !project.hasProperty("circuit.noCompose")
   plugins.withType<KotlinBasePlugin> {
     val isMultiPlatformPlugin = this is AbstractKotlinMultiplatformPluginWrapper
Index: settings.gradle.kts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/settings.gradle.kts b/settings.gradle.kts
--- a/settings.gradle.kts	(revision 06431d721baafe6d5839e36acc396721bd55bb73)
+++ b/settings.gradle.kts	(date 1692281701876)
@@ -49,6 +49,7 @@
         content { includeGroupByRegex("androidx.*") }
       }
     }
+    maven("https://maven.pkg.jetbrains.space/kotlin/p/kotlin/dev/")
 
     if (hasProperty("circuit.config.enableMavenLocal")) {
       mavenLocal()
@@ -127,6 +128,8 @@
       }
     }
 
+    maven("https://maven.pkg.jetbrains.space/kotlin/p/kotlin/dev/")
+
     if (hasProperty("circuit.config.enableMavenLocal")) {
       mavenLocal()
     }
Index: samples/counter/apps/gradle.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>ISO-8859-1
===================================================================
diff --git a/samples/counter/apps/gradle.properties b/samples/counter/apps/gradle.properties
--- a/samples/counter/apps/gradle.properties	(revision 06431d721baafe6d5839e36acc396721bd55bb73)
+++ b/samples/counter/apps/gradle.properties	(date 1692281570507)
@@ -1,3 +1,3 @@
 # Basically the JS equivalent of a fat jar
 # TODO retry without this after 1.9.10 https://youtrack.jetbrains.com/issue/KT-60852
-kotlin.js.ir.output.granularity=whole-program
\ No newline at end of file
+#kotlin.js.ir.output.granularity=whole-program
\ No newline at end of file

```

**Comment by sunildhiman90** [comment]
Hi @eymar , i am also facing this issue, As mentioned, i applied these changes:

1. kotlin version changed to:  1.9.10-456
2. applied this check in project build.gradle: 
```
afterEvaluate {
    project.extensions.findByType<org.jetbrains.compose.ComposeExtension>()?.also {
        it.kotlinCompilerPlugin.set("1.5.0")
        it.kotlinCompilerPluginArgs.add("suppressKotlinVersionCompatibilityCheck=1.9.10-456")
    }
}
```
3.  `-kotlin.js.ir.output.granularity=whole-program`, this line was already not there in gradle.properties

After making all these changes if i try to run its not working, i m getting following:
`Caused by: java.lang.RuntimeException: Compose Multiplatform 1.5.0 doesn't support Kotlin 1.9.10-456`

So which compose version we will have to use with this kotlin version?

**Comment by eymar** [comment]
This was fixed in kotlin 1.9.10 which is already available.
Compose Multiplatform 1.5.1 supports kotlin 1.9.10 out of a box

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-3486">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
